frontend/public/components/nav/section.tsx

```diff
@@ -41,158 +40,162 @@ const mergePluginChild = (
 };
 
 export const NavSection = connect(navSectionStateToProps)(
-  class NavSection extends React.Component<Props, NavSectionState> {
-    public state: NavSectionState;
+  withExtensions(isNavItem, isPerspective)(
```

Comment:
Can we _also_ support named props like this so that we can avoid `extensions.filter(isNavItem)` 
```suggestion
  withExtensions({ navItemExtensions: isNavItem, perspectiveExtensions: isPerspective })(
```

Reply:
Good idea! We can do that.

Reply:
But no need to hold this PR up on this request. It can be a followup enhancement.

Reply:
@christianvogt I've managed to do this as part of the PR :smiley: see commit https://github.com/openshift/console/pull/3383/commits/cbd0ba60e38c74980168fc9cfe8cf13fa709863d